src/components/ReviewInfoText/index.tsx

```diff
@@ -42,17 +48,39 @@ export const ReviewInfoText = ({
 
   const transactionsToGo = safeNonceNumber - nonce
 
+  useEffect(() => {
+    const fetchRecommendedNonce = async () => {
+      try {
+        const recommendedNonce = (await getRecommendedNonce(safeAddress)).toString()
+        setRecommendedNonce(recommendedNonce)
+      } catch (e) {
+        return
+      }
+    }
+    fetchRecommendedNonce()
+  }, [safeAddress])
+
+  const warningMessage = () => {
+    if (transactionsToGo < 0) {
```

Comment:
Btw I would put all warning related logic in this function.
Namely
```
  const isTxNonceOutOfOrder = () => {
    if (safeNonceNumber === nonce) return false
    if (lastTxNonce !== undefined && safeNonceNumber === lastTxNonce + 1) return false
    return true
  }
  const shouldShowWarning = isTxNonceOutOfOrder()

  const transactionsToGo = safeNonceNumber - nonce
```
The code above is needed only for the warning, and could be encapsulated there.

Reply:
The `shouldShowWarning` is needed for the conditional rendering so we need to call it before invoking `warningMessage()` but the
```
const transactionsToGo = safeNonceNumber - nonce
```
I will move inside of the `warningMessage`